src/Craftify.Geometry.Domain/LineSegmentIntersections.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace Craftify.Geometry.Domain;
+
+public static class LineSegmentIntersections
+{
+    public static bool Intersects(this LineSegment main, LineSegment other, out Point3D? intersectionPoint)
```

Comment:
To be honest, I don’t like using out, to me,  returning everything in returns make it cleaner,

I would use a given return type:

```csharp
public static (bool DoesIntersect, Point3D Point) Intersects(
        this LineSegment main,
        LineSegment other
    ) { }
```

This way If I use it in a chain (LINQ), it will be easier to filter, for example:

```csharp
LineSegment[] lines = [Line.ByStartPointAndEndPoint(Point.Origin(), Point.Origin())];
var targetLine = Line.ByStartPointAndEndPoint(Point.Origin(), Point.Origin());
var intersectedPoints = lines
    .Select(line => line.Intersects(targetLine))
    .Where(line => line.DoesIntersect)
    .Select(line => line.Point);
```

If we use out it would look like this:

```csharp
public static bool Intersects(
        this LineSegment main,
        LineSegment other,
        out Point3D? point
    ) { }
    
var intersectedPoints = lines
            .Where(line => line.Intersects(targetLine, out _))
            .Select(line =>
            {
                line.Intersects(targetLine, out var resultPoint);
                return resultPoint;
            });
```

Which makes us perform computation twice.

If we don’t want to perform computation twice we would need to create tuple inside LINQ:

```csharp
     var intersectedPoints = lines
            .Select(line =>
            {
                var doesIntersect = line.Intersects(targetLine, out var point);
                return (DoesIntersect: doesIntersect, Point: point);
            })
            .Where(result => result.DoesIntersect)
            .Select(result => result.Point);
```